e2e/chunked_test.go

```diff
@@ -98,3 +98,33 @@ func TestE2EChunked_BinderFast(t *testing.T) {
 		},
 	}))
 }
+
+func TestE2EChunked_ResponseReader(t *testing.T) {
+	const chars = "abcdefghijklmnopqrstuvwxyz"
+	mux := http.NewServeMux()
+	mux.HandleFunc("/test", func(w http.ResponseWriter, r *http.Request) {
+		b := make([]byte, 1000000)
+		for i := range b {
+			b[i] = chars[i%26]
+		}
+		w.Header().Add("Content-Type", "text/plain; charset=utf-8")
+		_, _ = w.Write(b)
+	})
```

Comment:
Let's make this handler to return infinite response. This way we'll be sure that nowhere in code we try to read response in memory when we call Reader.

In test we will read only part of the infinite response of course, like the first megabyte.